circuit/types/string/src/helpers/from_bits.rs

```diff
@@ -21,25 +21,73 @@ impl<E: Environment> FromBits for StringType<E> {
 
     /// Initializes a new string from a list of little-endian bits *with* trailing zeros (to byte-alignment).
     fn from_bits_le(bits_le: &[Self::Boolean]) -> Self {
+        // Determine the mode.
+        let mode = Mode::combine(bits_le[0].eject_mode(), bits_le.iter().skip(1).map(|bit| bit.eject_mode()));
+
+        // Inject the string size in bytes.
+        let size_in_bytes = Self::inject_size_in_bytes(mode, bits_le);
+
         // Ensure the list of booleans is byte-aligned.
```

Comment:
We can push the construction of the struct, and the computation of the `mode` also to the helper function. The only part that depends on endianness is the call to `U8::from_bits_{le,be}`, but we can pass that as a closure.